sssom/cli.py

```diff
@@ -31,77 +56,149 @@ def main(verbose, quiet):
+                   'the SSSOM default curie map. Must be one of metadata_only, sssom_default_only, merged')
+@output_option
+def parse(input: str, input_format: str, metadata: str, curie_map_mode: str, output: str):
+    """Parses a file in one of the supported formats (such as obographs) into an SSSOM TSV file.
+
+    Args:
+
+        input (str): The path to the input file in one of the legal formats, eg obographs, aligmentapi-xml
+        input_format (str): The string denoting the input format.
+        metadata (str): The path to a file containing the sssom metadata (including curie_map) to be used during parse.
+        curie_map_mode (str): Curie map mode.
+        output (str): The path to the SSSOM TSV output file.
+
+    Returns:
+    
+        None.
     """
-    parse file (currently only supports conversion to RDF)
+
+    parse_file(input_path=input, output_path=output, input_format=input_format, metadata_path=metadata,
+               curie_map_mode=curie_map_mode)
+
+
+@main.command()
+@input_option
+def validate(input: str):
+    """Takes 1 sssom file as input and produce an error report
+
+    Args:
+
+        input (str): Input file. For e.g.: SSSOM tsv file
+
+    Returns:
+
+        None.
     """
-    parse_file(input_path=input, output_path=output, input_format=input_format, metadata_path=metadata, curie_map_mode=curie_map_mode)
+
+    validate_file(input_path=input)
 
 
 @main.command()
-@click.option('-i', '--input', required=True, type=click.Path())
-@click.option('-d', '--output-directory', type=click.Path())
+@input_option
+@output_directory_option
 def split(input: str, output_directory: str):
+    """Parse file (currently only supports conversion to RDF)
+
+    Args:
+
+        input (str): Input file. For e.g.: SSSOM tsv file.
+        output_directory (str): Output directory path.
+
+    Returns:
+
+        None.
     """
-    parse file (currently only supports conversion to RDF)
-    """
+
     split_file(input_path=input, output_directory=output_directory)
 
 
 @main.command()
-@click.option('-W', '--inverse-factor')
-@click.argument('input')
-def ptable(input, inverse_factor):
-    """
-    write ptable (kboom/boomer input)
-    should maybe move to boomer (but for now it can live here, so cjm can tweak
+@input_option
+@click.option('-W', '--inverse-factor', help='Inverse factor.')
+def ptable(input: str, inverse_factor: str):
+    """Write ptable (kboom/boomer input) should maybe move to boomer (but for now it can live here, so cjm can tweak
+
+    Args:
+
+        input (str): Input file. For e.g.: SSSOM tsv file.
+        inverse_factor (str): Inverse factor.
+
+    Returns:
+
+        None
+
     """
-    df = parse(input)
-    df = collapse(df)
+    msdf = from_tsv(input)
+    # df = parse(input)
+    df = collapse(msdf.df)
     # , priors=list(priors)
     rows = dataframe_to_ptable(df)
     for row in rows:
         logging.info("\t".join(row))
```

Comment:
I guess I am responsible for this... what was I thinking. This isn't logging output, it's the primary output

Reply:
This is my mistake, not yours.. I want to generally be able to stream outputs from all commands in that way; what is the best way to do this then - using print() in conjunction with a stream (either a file output stream or stdout)?

Reply:
@hrshdhgd I will fix this

Reply:
for ptable we are now using print, but I want to ensure with the architecture so will review with @cmungall on Monday and merge.